### PR TITLE
Support @Key annotations on JAX-RS resource method parameters

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: jaxrs/src/main/java/com/proofpoint/jaxrs/TimingResourceDynamicFeature.java [hunk 1]
```diff
@@ -100,4 +129,68 @@ else if (type.isAnnotationPresent(Path.class)) {
...
+
+                            synchronized (registeredMap) {
+                                SparseTimeStat existingStat = registeredMap.get(key);
+                                if (existingStat != null) {
+                                    reinsertedSet.add(existingStat);
+                                    return existingStat;
+                                }
+                                registeredMap.put(key, returnValue);
+                                Builder<String, String> tagBuilder = ImmutableMap.builder();
+                                for (int i = 0; i < keyNames.size(); ++i) {
```
Comment (gwittel) [inline jaxrs/src/main/java/com/proofpoint/jaxrs/TimingResourceDynamicFeature.java]: General clarification -- Am I correct in reading this that the somewhat opaque detail is that `keyNames` and `key` are both derived from the data created by the bytebuddy code generation?  
Therefore the assumption here that `keyNames` and `key` will always have same length and ordering.

This behavior relies on code in `TimingFilter` generating the tag/cache lookups the same way as the `TimingWrapper` generates the data used to populate the loadingcache.

Do we need tests to cover the scenario where only one side is updated (say a new default metric) causing a length mismatch?

Comment (johngmyers) [inline jaxrs/src/main/java/com/proofpoint/jaxrs/TimingResourceDynamicFeature.java]: `CacheImplementation` came in the refactor prior to adding ByteBuddy.

But there is an invariant that the keyNames produced by `TimingResourceDynamicFeature.configure(...)` to create the `LoadingCache` match in number and order the values given to the `LoadingCache` by `TimingFilter.filter(...)`.

I don't see a useful test for this particular point. It is an implementation detail.

Comment (johngmyers) [inline jaxrs/src/main/java/com/proofpoint/jaxrs/TimingResourceDynamicFeature.java]: To clarify, the existing tests of the API, which check that particular metrics are produced for particular calls, would catch such a mismatch.

**Comment by johngmyers** [comment]
Rebased and merged

**Pull request closed without merging**